pkg/virt-handler/vm.go

```diff
@@ -1108,6 +1118,13 @@ func (d *VirtualMachineController) updateVMIStatus(origVMI *v1.VirtualMachineIns
 	vmi := origVMI.DeepCopy()
 	oldStatus := *vmi.Status.DeepCopy()
 
+	// Update existing VMIs topologyHints if needed
+	if vmi.Status.TopologyHints == nil {
+		if topologyHints, _, _ := d.topologyHinter.TopologyHintsForVMI(vmi); topologyHints != nil {
+			vmi.Status.TopologyHints = topologyHints
+		}
+	}
```

Comment:
I see that virt-controller does:
```go
if vmi.Status.TopologyHints == nil && c.topologyHinter.TopologyHintsRequiredForVMI(vmi) {
	if topologyHints, err := c.topologyHinter.TopologyHintsForVMI(vmi); err != nil {
		c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, FailedGatherhingClusterTopologyHints, err.Error())
		return &syncErrorImpl{err, FailedGatherhingClusterTopologyHints}
	} else if topologyHints != nil {
		vmiCopy.Status.TopologyHints = topologyHints
	}
}
```

WDYT about adding `c.topologyHinter.TopologyHintsRequiredForVMI(vmi)` to here as well?

Reply:
I don't think it's needed, `TopologyHintsForVMI()` internally calls `TopologyHintsRequiredForVMI()` and just returns nil in case it's not required, we can simply avoid calling it twice.

Actually maybe this can be simplified in main too.

Reply:
Makes sense, thanks!